lib/SILGen/SILGenPoly.cpp

```diff
@@ -399,6 +399,12 @@ ManagedValue Transform::transform(ManagedValue v,
     inputOTK = OTK_None;
   }
 
+  if (inputOTK != OTK_None && outputOTK == OTK_None) {
+    SILValue result = SGF.B.createUncheckedBitCast(Loc, v.getValue(),
```

Comment:
Probably, anyway there's more ambiguity issues after switching constraint to `Subtype`, it will take me some time to figure this out, I'll make an update when ready.